youtube_dl/extractor/spankbang.py

```diff
@@ -94,3 +95,40 @@ def _real_extract(self, url):
             'formats': formats,
             'age_limit': age_limit,
         }
+
+
+class SpankBangPlaylistIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:(?:www|m|[a-z]{2})\.)?spankbang\.com/(?P<id>[\da-z]+)/playlist'
+    _TEST = {
+        'url': 'https://spankbang.com/ug0k/playlist/big+ass+titties',
+        'info_dict': {
+            'id': 'ug0k',
+            'title': 'Big Ass Titties playlist',
+        },
+        'playlist_mincount': 2,
+    }
+
+    def _extract_entries(self, webpage, id):
+        video_items = re.findall(r'<div[^>]+class=[\'"].*?video-item[^>]*>\s*(.+?)>', webpage)
+
+        entries = []
+        if video_items:
+            for div in video_items:
+                page_url = self._search_regex(
+                    r'href="/?(' + id + '-[\da-z]+/playlist/[^"]+)', div, 'page url', default=None)
+
+                if page_url:
+                    page = self._download_webpage(urljoin('http://spankbang.com', page_url), id)
```

Comment:
No. It's a job if the video extractor.

Reply:
This is needed to get the canonical url for the single video extractor [SpankBang] to use

Reply:
Again: video URLs are already available on playlist page.

Reply:
these urls are in the for /playlist_id-playlist_item_id/playlist/playlistname
even if you replace playlist with video and only use the playlist_item_id it does point to the the actual video url

Reply:
*does not

Reply:
I'm not talking about **these** URLs.

Reply:
Where are they